internal/deb/extract.go

```diff
@@ -245,22 +222,60 @@ func extractData(dataReader io.Reader, options *ExtractOptions) error {
 					return err
 				}
 			}
+			link := tarHeader.Linkname
+			if tarHeader.Typeflag == tar.TypeLink {
+				// A hard link requires the real path of the target file.
+				link = filepath.Join(options.TargetDir, link)
```

Comment:
Can we please add a TODO here reflecting the agreement:
```
// TODO This path must be sanitized to prevent a crafted tarball from referring to content
// outside the target directory.  This issue also affects targetPath, but it's more relevant
// here as this is importing external content into the target directory. This is not a more
// serious issue because these tarballs are signed official packages, but must be fixed
// regardless.
```